homeassistant/components/switchbot_cloud/config_flow.py

```diff
@@ -0,0 +1,66 @@
+"""Config flow for SwitchBot via API integration."""
+
+from logging import getLogger
+from typing import Any
+
+from switchbot_api import CannotConnect, InvalidAuth, SwitchBotAPI
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_KEY, CONF_API_TOKEN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN, ENTRY_TITLE
+
+_LOGGER = getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_API_TOKEN): str,
+        vol.Required(CONF_API_KEY): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    await SwitchBotAPI(
+        token=data[CONF_API_TOKEN], secret=data[CONF_API_KEY]
+    ).list_devices()
+    return data
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class SwitchBotCloudConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for SwitchBot via API."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            try:
+                info = await validate_input(self.hass, user_input)
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
```

Comment:
Please also log the exception. This way if this ever throws you're not left with people who dont see anything useful in the logs making it unable for you to solve the issue.

Reply:
With the following code:
```python
              try:
                raise Exception("LOL")
                info = await validate_input(self.hass, user_input)
            except CannotConnect:
                errors["base"] = "cannot_connect"
            except InvalidAuth:
                errors["base"] = "invalid_auth"
            except Exception:  # pylint: disable=broad-except
                _LOGGER.exception("Unexpected exception")
                errors["base"] = "unknown"
```

It already logs:
```log
2023-09-10 11:13:22.195 ERROR (MainThread) [homeassistant.components.switchbot_cloud.config_flow] Unexpected exception
Traceback (most recent call last):
  File "/workspaces/home-assistant-switchbot-via-api/homeassistant/components/switchbot_cloud/config_flow.py", line 49, in async_step_user
    raise Exception("LOL")
Exception: LOL
```

Reply:
Huh what, I did not know this was a feature

Reply:
@joostlek Just to be sure, the "LOL" was not me being sarcastic or anything, it is just something I am mechanically used to write lol